src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use DateTime;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    public function findActivePromotionsByPid(Pid $pid, ?int $limit, int $offset): array
```

Comment:
is a repository. Array of objects. You can see the hydration method at the end of the function:
```return $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);```